solrmonitor/zkdispatcher.go

```diff
@@ -120,19 +117,17 @@ func (d *ZkDispatcher) eventLoop() {
 	defer atomic.AddInt32(&d.runningProcs, -1)
 
 	for {
-		chosen, recv, recvOK := reflect.Select(d.selectCases)
+		chosen, recv, _ := reflect.Select(d.selectCases)
```

Comment:
btw, I was thinking about this problem, and I think the reason we're getting channel full is that the select here is randomish over all the available channels-- the newHandler channel doesn't have any particular priority.  So in a situation where tons of events all come in at one and tons of channels are readable, it's basically random whether the newHandler channel gets called enough to drain the incoming queue.


I might suggest the following macro structure:

```
for {

  for {
    	select {
	case newHandler := <- d. newHandlerChan:
		// queue up the new handler
	default:
		break // the loop
	} 
  }

  // existing code with dynamic select
}
```

That way, every time you handle a dynamic select, you drain out the incoming queue before proceeding.